# Actividad Practica 1

La presente actividad tiene como objetivo la obtención de 3 datasets potenciales para realizar el proyecto final del curso de Data Science I de Coderhouse. Las características requeridas de cada dataset son:

* 15 columnas mínimo
* 2000 filas mínimo

Los datasets seleccionados que se describirán en las secciones siguientes son:

1. **household_income**: Dataset que muestra la información de varias variables que pueden afectar el resultado de la renta anual de una familia. Puede encontrar el dataset en el siguiente [Link](https://www.kaggle.com/datasets/stealthtechnologies/regression-dataset-for-household-income-analysis).
2. **spotify_songs_2024**: Dataset que muestra la información del desempeño de canciones alojadas en spotify y tambien en otras plataformas, en términos de escuchas, vistas, playlists, menciones, etc. Puede encontrar el dataset en el siguiente [Link](https://www.kaggle.com/datasets/nelgiriyewithana/most-streamed-spotify-songs-2024).
3. **athlete_events.csv**: Dataset que muestra la información de resultados individuales de todos los atletas olimpicos desde el inicio de los juegos olímpicos modernos, tanto de verano como de invierno. Puede encontrar el dataset en el siguiente [Link](https://www.kaggle.com/datasets/heesoo37/120-years-of-olympic-history-athletes-and-results).

## Importación de librerias

In [1]:
import pandas as pd

# Dataset #1: *household_income*

## Descripción del dataset

Corresponde a un dataset con varios factores demográficos y socioeconomicos que influencian el ingreso familiar anual. Las columnas que componen el dataset se describen a continuación.

* Age: Edad del jefe de hogar (18 to 70 years).

* Education Level: Nivel de educación más alto (High School, Bachelor's, Master's, Doctorate).

* Occupation: Tipo de ocupación (Healthcare, Education, Technology, Finance, Others).

* Number of Dependents: Numero de personas dependientes del jefe de hogar (0 to 5).

* Location: Tipo de zona residencial (Urban, Suburban, Rural).

* Work Experience: Años de experiencia laboral (0 to 50 years).

* Marital Status: Estado civil del jefe de hogar (Single, Married, Divorced).

* Employment Status: Situacion laboral del jefe de hogar (Full-time, Part-time, Self-employed).

* Household Size: Numero total de miembros del hogar (1 to 7).

* Homeownership Status: Situación de posesión de la vivienda (Own, Rent).

* Type of Housing: Tipo de vivienda (Apartment, Single-family home, Townhouse).

* Gender: Género del jefe de hogar (Male, Female).

* Primary Mode of Transportation: Modo de transporte del jefe de hogar (Car, Public transit, Biking, Walking).

* Annual Household Income: Ingreso anual del hogar en USD

## Importación del dataset

In [2]:
df = pd.read_csv("datasets/household_income.csv")
df.head()

Age Education_Level  Occupation  Number_of_Dependents Location  \
0   56        Master's  Technology                     5    Urban   
1   69     High School     Finance                     0    Urban   
2   46      Bachelor's  Technology                     1    Urban   
3   32     High School      Others                     2    Urban   
4   60      Bachelor's     Finance                     3    Urban   

   Work_Experience Marital_Status Employment_Status  Household_Size  \
0               21        Married         Full-time               7   
1                4         Single         Full-time               7   
2                1         Single         Full-time               7   
3               32        Married         Full-time               1   
4               15        Married     Self-employed               4   

  Homeownership_Status     Type_of_Housing  Gender  \
0                  Own           Apartment    Male   
1                  Own           Apartment    Male   
2                  Own  Single-family home  Female   
3                  Own           Apartment  Female   
4                  Own           Townhouse    Male   

  Primary_Mode_of_Transportation  Income  
0                 Public transit   72510  
1                         Biking   75462  
2                            Car   71748  
3                            Car   74520  
4                        Walking  640210

In [3]:
df.shape

(10000, 14)

## Descripción de las variables principales

### Información y tipos de datos

El dataset se compone de 14 columnas y 10000 entradas no nulas. Éste dataset ya se encuentra limpio y los tipos de datos corresponden a la información proporcionada en cada columna.

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Age                             10000 non-null  int64 
 1   Education_Level                 10000 non-null  object
 2   Occupation                      10000 non-null  object
 3   Number_of_Dependents            10000 non-null  int64 
 4   Location                        10000 non-null  object
 5   Work_Experience                 10000 non-null  int64 
 6   Marital_Status                  10000 non-null  object
 7   Employment_Status               10000 non-null  object
 8   Household_Size                  10000 non-null  int64 
 9   Homeownership_Status            10000 non-null  object
 10  Type_of_Housing                 10000 non-null  object
 11  Gender                          10000 non-null  object
 12  Primary_Mode_of_Transportation  10000 non-null 

### Descripción estadistica del dataset

In [5]:
# Descripción de los datos numericos
df.describe()

Age  Number_of_Dependents  Work_Experience  Household_Size  \
count  10000.000000          10000.000000     10000.000000    10000.000000   
mean      44.021700              2.527000        24.858800        3.989600   
std       15.203998              1.713991        14.652622        2.010496   
min       18.000000              0.000000         0.000000        1.000000   
25%       31.000000              1.000000        12.000000        2.000000   
50%       44.000000              3.000000        25.000000        4.000000   
75%       57.000000              4.000000        37.000000        6.000000   
max       70.000000              5.000000        50.000000        7.000000   

             Income  
count  1.000000e+04  
mean   8.168382e+05  
std    1.821089e+06  
min    3.104400e+04  
25%    6.844600e+04  
50%    7.294300e+04  
75%    3.506675e+05  
max    9.992571e+06

In [7]:
df["Income"].median()

72943.0

In [6]:
# Descripción de los datos no-numéricos
df.describe(exclude="number")

Education_Level  Occupation Location Marital_Status Employment_Status  \
count            10000       10000    10000          10000             10000   
unique               4           5        3              3                 3   
top         Bachelor's  Healthcare    Urban        Married         Full-time   
freq              4058        3035     7037           5136              5004   

       Homeownership_Status     Type_of_Housing Gender  \
count                 10000               10000  10000   
unique                    2                   3      2   
top                     Own  Single-family home   Male   
freq                   6018                4055   5123   

       Primary_Mode_of_Transportation  
count                           10000  
unique                              4  
top                    Public transit  
freq                             4047

### Observaciones sobre la información contenida en el dataset

Se pueden recoger las siguientes observaciones del dataset, en esta etapa preliminar.

- Existen 13 variables que se pueden relacionar con el ingreso, siendo 4 numéricas y 9 categóricas
- La edad de los encuestados se encuentra entre los 18 y los 70 años, con un promedio de **44.02**
- La cantidad de dependientes se encuentra entre 0 y 5 personas con un promedio de **2.52**
- La experiencia laboral se encuentra entre 0 y 50 años, con un promedio de **24.85**
- La cantidad total de integrantes se encuentra entre 1 y 7 personas, con un promedio de **3.98**
- El ingreso anual se encuentra entrelos 31044 USD y los 9992571 USD anuales, con un promedio de **816838 USD**. La mediana de esta variables es **72943 USD**.